internals-js/src/extractSubgraphsFromSupergraph.ts

```diff
@@ -499,11 +499,27 @@ function removeNeedlessProvides(subgraph: Subgraph) {
   }
 }
 
+function isExternalOrHasExternalImplementations(field: FieldDefinition<CompositeType>): boolean {
+  if (field.hasAppliedDirective(externalDirectiveName)) {
+    return true;
+  }
+  const parentType = field.parent!;
```

Comment:
> This is obviously a much larger discussion than this PR

Yes :). And for the record, I don't love this and that is one of the thing I'd change in the whole `Schema` hierarchy, but this is the result of running out of time when designing that whole thing and needing to move to more pressing matters. Definitively happy to discuss this further and try to improve it, just not here :).

> how do we know that we can assert that parent exists here?

We kind of don't in absolute terms, but in practice we know we shouldn't pass an incomplete schema to this method. To be fair, the failure mode here is pretty ugly and it might be cleaner (while being a fairly small change) to make `parent` a getter that never return `undefined` but instead assert the parent is set, as we'd at least throw early if this end up misused.

> why do we allow for SchemaElements to remove their parents?

It's more that the whole `Schema` abstraction is mutable and we can remove elements, and when an element is removed we clear its parent link to avoid some kind of half-broken state (essentially, trying to ensure that if code mistakenly continue to use some removed element, this behave a bit more predictably).